partitura/io/exportkern.py

```diff
@@ -0,0 +1,327 @@
+# -*- coding: utf-8 -*-
+"""
+This module contains methods for exporting Kern files.
+"""
+import math
+from collections import defaultdict
+
+import numpy
+
+import partitura.score as spt
+from operator import itemgetter
+from typing import Optional
+import numpy as np
+import warnings
+from partitura.utils import partition, iter_current_next, to_quarter_tempo
+from partitura.utils.misc import deprecated_alias, PathLike
+
+__all__ = ["save_kern"]
+
+
+ACC_TO_SIGN = {
+    0: "n",
+    -1: "-",
+    1: "#",
+    -2: "--",
+    2: "##",
+}
+
+# Kern notes encoding has a dedicated octave for each note.
+KERN_NOTES = {
+    ("C", 3): "C",
+    ("D", 3): "D",
+    ("E", 3): "E",
+    ("F", 3): "F",
+    ("G", 3): "G",
+    ("A", 3): "A",
+    ("B", 3): "B",
+    ("C", 4): "c",
+    ("D", 4): "d",
+    ("E", 4): "e",
+    ("F", 4): "f",
+    ("G", 4): "g",
+    ("A", 4): "a",
+    ("B", 4): "b",
+}
+
+KERN_DURS = {
+    "maxima": "000",
+    "long": "00",
+    "breve": "0",
+    "whole": "1",
+    "half": "2",
+    "quarter": "4",
+    "eighth": "8",
+    "16th": "16",
+    "32nd": "32",
+    "64th": "64",
+    "128th": "128",
+    "256th": "256",
+}
+
+KEYS = ["f", "c", "g", "d", "a", "e", "b"]
+
+
+class KernExporter(object):
+    """
+    Class for exporting a partitura score to Kern format.
+
+    Parameters
+    ----------
+    part: spt.Part
+        Part to export to Kern format.
+    """
+
+    def __init__(self, part):
+        self.part = part
+        note_array = part.note_array(include_staff=True)
+        num_measures = len(part.measures)
+        num_notes = len(part.notes)
+        num_rests = len(part.rests)
+        self.unique_voc_staff = np.unique(note_array[["voice", "staff"]], axis=0)
+        self.vocstaff_map_dict = {
+            f"{self.unique_voc_staff[i][0]}-{self.unique_voc_staff[i][1]}": i
+            for i in range(self.unique_voc_staff.shape[0])
+        }
+        # Part elements is really the maximum number of lines we could have in the kern file
+        # we add some to account for the **kern and the *- encoding at beginning and end of file and also tandem elements
+        # that might be added. We also add the number of measures to account for the measure encoding
+        total_elements_ish = num_measures + num_notes + num_rests + 2 + 10
+        self.out_data = np.empty(
+            (total_elements_ish, len(self.unique_voc_staff)), dtype=object
+        )
+        self.unique_times = np.array([p.t for p in part._points])
+        # Fill all values with the "." character to filter afterwards
+        self.out_data.fill(".")
+        self.out_data[0] = "**kern"
+        self.out_data[-1] = "*-"
+        # Add the staff element to the second line
+        for i in range(self.unique_voc_staff.shape[0]):
+            self.out_data[1, i] = f"*staff{self.unique_voc_staff[i][1]}"
+        self.prev_note_time = None
+        self.prev_note_col_idx = None
+        self.prev_note_row_idx = None
+
+    def parse(self):
```

Comment:
what is a `parse` function doing in an export? Some doc for this function will help a lot!